modules/samples/product/Chad/main_www/js/WSRequest.js

```diff
@@ -460,9 +459,9 @@ WSRequest.util = {
             return 'ie7';
         } else if (ua.indexOf('msie') != -1) { // IE
             return 'ie';
-        } else if (ua.indexOf('safari') != -1) { // Safari (check before Gecko because it includes "like Gecko")
-            return 'safari';
-        } else if (ua.indexOf('gecko') != -1) { // Gecko
+        } else if (ua.indexOf('webkit') != -1) { // Webkit supported agent
+            return 'webkit';
+        } else if (ua.indexOf('gecko') != -1) { // Gecko supported agent
```

Comment:
check for the possibility of introducing a switch case statement here instead of the else ifs.